### PR TITLE
Mawilkie

### PR DESCRIPTION
**Pull request by markwilkie**
Added a roadmap and updated summary with requirements and principles.

### CODE REVIEW

File: Documentation/Overview.md [hunk 1]
```diff
@@ -9,21 +9,22 @@ The primary concept is to break the infrastructure into “pay for play” compo
...
--  The assumption is that the ProdCon V2 effort will be largely addressing the repo level contracts themselves.  As mentioned earlier, the intent of this effort is to focus on the tools.  Note that there is likely some overlap (like with bootstrapping), but we'll deal with those as they come up.
--  We like to take advantage of the ProdCon V2 effort to start off well.  To that end, there is some urgency so that we can be ready.
+### Principles
+- Updates and changes should always be done with all the repos (not just yours) in mind.  This implies compromise by all to achieve a better common goal.
+- Simplicity and austerity are our friends.  We want to avoid clever, magic, or fancy features.
+- Incremental updates where ever humanly possible.  (avoid big changes with a "switch" when possible)
+- When something needs to be done across multiple repos, the extra work required to put the feature in Arcade is worth it.
+- Where appropriate, testing should be in one repo before "graduating" to Arcade.  This way we can learn more what's needed, thus minimizing churn.
```
Comment (dagood) [inline Documentation/Overview.md]: I couldn't find an issue about getting Arcade to support this, so I filed https://github.com/dotnet/arcade/issues/96.

Comment (weshaggard) [reply Documentation/Overview.md]: I think we need to be careful about having repo's build and consume these tools/tasks in their own repo, I added a note to that affect in the issue. I would much prefer a model where folks add the tasks in arcade as optional packages that can be consumed and then if necessary prompted to required for the core tools sdk.

File: Documentation/Overview.md [hunk 2]
```diff
@@ -47,14 +48,9 @@ This document speaks _only_ to the first bullet point below - e.g. tasks/targets
...
--  There is one (separate from product) feed for all common infrastructure (this is probably correct, but needs to be thought through. (https://github.com/dotnet/core-eng/pull/2552)   For example, what about our community?  Also, private repos?)
--  A method of discovering what shared infra offerings are available is important to figure out.  Likely this is largely a documentation effort, but perhaps there can be a special service which helps out.  Possibly use the "gallery" feature as an aid?  Documentation for each package will help too.
--  Working out the right way to “fork” infra (in this case packages) with the product branch needs additional attention.  The current thinking is to embed the git hash into every binary/package produced, and then ensure the repo is buildable clean given a specific hash.
-- Package reference should **not** float.  This implies the need for a "Maestro" like tool (re-use the prodcon approach?) to explicitely update tool references in multiple repos.
+### Arcade building repos Requirements
+- Arcade builds, tests, packages, signs, and publishes itself using itself
```
Comment (weshaggard) [inline Documentation/Overview.md]: I would add "using itself and the shipping sdk/cli".

File: Documentation/Overview.md [hunk 3]
```diff
@@ -47,14 +48,9 @@ This document speaks _only_ to the first bullet point below - e.g. tasks/targets
...
+- All tools bootstrapped in, getting as close as technically possible to 'clone and build' on a clean machine
+- A repo level API is explicitly defined and implemented, not just implied 
+- Method exist to directly manage Arcade in each participating repo
```
Comment (weshaggard) [inline Documentation/Overview.md]: by manage to you mean "maintain and update" in each repo?

File: Documentation/Overview.md [hunk 4]
```diff
@@ -9,21 +9,22 @@ The primary concept is to break the infrastructure into “pay for play” compo
...
 - "Resources" as planned by VSTS.
 
-### Approach
--  The idea is to start with the "satellite" (non core) build tools so we can learn what works and what doesn't.  Once we're more comfortable, we can start to move farther into the "core" (touchier) tools.
--  The assumption is that the ProdCon V2 effort will be largely addressing the repo level contracts themselves.  As mentioned earlier, the intent of this effort is to focus on the tools.  Note that there is likely some overlap (like with bootstrapping), but we'll deal with those as they come up.
--  We like to take advantage of the ProdCon V2 effort to start off well.  To that end, there is some urgency so that we can be ready.
+### Principles
+- Updates and changes should always be done with all the repos (not just yours) in mind.  This implies compromise by all to achieve a better common goal.
+- Simplicity and austerity are our friends.  We want to avoid clever, magic, or fancy features.
+- Incremental updates where ever humanly possible.  (avoid big changes with a "switch" when possible)
```
Comment (weshaggard) [inline Documentation/Overview.md]: I would add "breaking changes" require either back compat switches or branching to make sure repo's have a path forward without needing to react to breaking changes, in particular servicing branches.

File: Documentation/Overview.md (hunk 1 above)
Comment (markwilkie) [reply Documentation/Overview.md]: I commented on #96 as well.

File: Documentation/Overview.md (hunk 4 above)
Comment (markwilkie) [reply Documentation/Overview.md]: Added

File: Documentation/Overview.md (hunk 2 above)
Comment (markwilkie) [reply Documentation/Overview.md]: done

File: Documentation/Overview.md (hunk 3 above)
Comment (markwilkie) [reply Documentation/Overview.md]: Yea, that's better.  Updated.

**Pull request merged**